lib/ddtrace/configuration/settings.rb

```diff
@@ -545,16 +545,35 @@ def initialize(*_)
 
         # TODO: This setting is not tracer-specific and should be moved to top-level or to the transport.
         option :port
-        option :priority_sampling # TODO: Deprecate
+
+        # Enables {https://docs.datadoghq.com/tracing/trace_retention_and_ingestion/#datadog-intelligent-retention-filter
+        # Datadog intelligent retention filter}.
+        # @default `true`
+        # @return [Boolean,nil]
+        option :priority_sampling
 
         # A custom sampler instance.
         # The object must respect the {Datadog::Sampler} interface.
         # @default `nil`
         # @return [Object,nil]
         option :sampler
-        option :transport_options, default: ->(_i) { {} }, lazy: true # TODO: Deprecate
-        option :writer # TODO: Deprecate
-        option :writer_options, default: ->(_i) { {} }, lazy: true # TODO: Deprecate
+
+        # A custom {Hash} with keyword options to be passed to the initializer of
+        # the tracer writer class.
+        # @default `{}`
+        # @return [Hash,nil]
+        option :transport_options, default: ->(_i) { {} }, lazy: true
+
+        # A custom writer instance.
+        # The object must respect the {Datadog::Writer} interface.
+        # @default `nil`
+        # @return [Object,nil]
+        option :writer
```

Comment:
Should we document this as for internal-use only? When would customers want/need to mess with this?

Reply:
I've seen a few cases where custom writers were used for users that wanted to test if they were instrumenting things correctly with `ddtrace`.

I can document here be dragons here.

Reply:
At some point it'd be nice to perhaps understand that use case more and provide something a bit nicer for those customers, but happy with the "here be dragons" for now :)